tests/phpunit/elementor/core/page-assets/test-loader.php

```diff
@@ -0,0 +1,202 @@
+				],
+			],
+			'scripts' => [
+				'e-script-add-asset' => [
+					'src' => 'e-script-add-asset-src',
+					'version' => 'e-script-add-asset-version',
+					'dependencies' => [],
+				],
+			],
+		];
+
+		$assets_loader->add_assets( $assets );
+
+		$added_assets = $assets_loader->get_assets();
+
+		foreach ( $assets as $assets_type => $assets_type_data ) {
+			$this->assertArrayHasKey( $assets_type, $added_assets, $assets_type . ' was not properly added to assets.' );
+
+			foreach ( $assets_type_data as $asset_name => $asset_data ) {
+				$this->assertArrayHasKey( $asset_name, $added_assets[ $assets_type ], $asset_name . ' was not properly added to ' . $assets_type . ' of assets.' );
+
+				foreach( $asset_data as $key => $value ) {
+					$this->assertArrayHasKey( $key, $added_assets[ $assets_type ][ $asset_name ], $key . ' was not properly added to ' . $asset_name );
+					$this->assertEquals( $value, $added_assets[ $assets_type ][ $asset_name ][ $key ] );
+				}
+			}
+		}
+	}
+
+	public function test_enqueue_assets() {
+		global $wp_styles;
+		global $wp_scripts;
+
+		$this->add_optimized_assets_loading_experiment();
+
+		// When "optimized assets loading" mode is inactive, all assets should be loaded by default.
+		$this->elementor()->experiments->set_feature_default_state( 'e_optimized_assets_loading', Experiments_Manager::STATE_INACTIVE );
+
+		$this->add_testing_assets( 'e-style-default-enqueue', 'e-script-default-enqueue' );
+
+		$assets_loader = $this->elementor()->assets_loader;
+
+		$assets_loader->enqueue_assets();
+
+		$this->assertContains( 'e-style-default-enqueue', $wp_styles->queue, 'e-style-default-enqueue was not properly enqueued.' );
+		$this->assertContains( 'e-script-default-enqueue', $wp_scripts->queue, 'e-script-default-enqueue was not properly enqueued.' );
+
+		// When "optimized assets loading" mode is active, assets should be loaded only when enabled.
+		$this->elementor()->experiments->set_feature_default_state( 'e_optimized_assets_loading', Experiments_Manager::STATE_ACTIVE );
+
+		$this->add_testing_assets( 'e-style-dynamic-enqueue', 'e-script-dynamic-enqueue' );
+
+		$assets_loader->enqueue_assets();
+
+		// Making sure that the assets are not enqueued before they are enabled.
+		$this->assertNotContains( 'e-style-dynamic-enqueue', $wp_styles->queue, 'e-style-dynamic-enqueue should not be enqueued before it\'s enabled.' );
+		$this->assertNotContains( 'e-script-dynamic-enqueue', $wp_scripts->queue, 'e-script-dynamic-enqueue should not be enqueued before it\'s enabled.' );
+
+		$assets_loader->enable_assets( [
+			'styles' => ['e-style-dynamic-enqueue'],
+			'scripts' => ['e-script-dynamic-enqueue'],
+		] );
+
+		$assets_loader->enqueue_assets();
+
+		$this->assertContains( 'e-style-dynamic-enqueue', $wp_styles->queue, 'e-style-dynamic-enqueue was not properly enqueued.' );
+		$this->assertContains( 'e-script-dynamic-enqueue', $wp_scripts->queue, 'e-script-dynamic-enqueue was not properly enqueued.' );
```

Comment:
Split each group of asserts to a different case

Reply:
Done.